### ISSUE TITLE
 router-module with keepDefaultRouter broken with 2.15

### ISSUE THREAD

**Issue by nachogarcia** [nuxt/nuxt.js#8831]
### Versions

- nuxt:  2.15.0
- node: 14.15.0

Probably regression created by https://github.com/nuxt/nuxt.js/pull/8520

### Reproduction

Updated the current project which has a custom router overriding the default.

`    ['@nuxtjs/router', { keepDefaultRouter: true }],`

![image](https://user-images.githubusercontent.com/9975630/107967672-b4e71300-6fad-11eb-99cb-3729b906dca1.png)

TypeError
Cannot read property 'app' of undefined

```
export function createRouter (ssrContext, config) {
  const base = (config.app && config.app.basePath) || routerOptions.base
```

**Comment by pi0** [comment]
Hi @nachogarcia. Thanks for report. I have filed https://github.com/nuxt-community/router-module/issues/91 to fix issue in module.

**Comment by thingsneverchange** [comment]
This error happened to me with Nuxt-Edge a while ago. (I build multiple websites using one single Nuxt instance.)
For others stuck with this error:

```
export async function createRouter(ssrContext, createDefaultRouter, routerOptions) {

   return new Router({
     ...routerOptions,
     routes: routerOptions.routes
   });

}
```
This seems to work with the 'keepDefaultRouter' enabled on Nuxt 2.15.

**createDefaultRouter(ssrContext).options** This part can be replaced with **routerOptions.options**. 

Please let me know if this trick is not universally done as I am using it on my production server.

**Comment by pi0** [comment]
should be fixed by 1.5.. Please consult docs for 2.15+ upgrade: https://github.com/nuxt-community/router-module#accessing-default-router
